pkg/provider/provider.go

```diff
@@ -38,3 +44,36 @@ type (
 		AppName string
 	}
 )
+
+// HandleProviderValidation ensures that the klotho configuration and construct graph are valid for the provider
+//
+// The current checks consist of:
+//   - types defined in klotho configuration for each construct is valid for the provider
+func HandleProviderValidation(p Provider, config *config.Application, constructGraph *core.ConstructGraph) error {
+
+	var errs multierr.Error
+	log := zap.L().Sugar()
+	for _, resource := range constructGraph.ListConstructs() {
+		if _, ok := resource.(*core.InternalResource); ok {
+			continue
+		}
+		resourceValid := false
+		mapping := p.GetKindTypeMappings(resource)
+		if len(mapping) == 0 {
+			errs.Append(errors.Errorf("Provider, %s, Does not support %s ", p.Name(), reflect.ValueOf(resource).Kind()))
+			continue
+		}
+		resourceType := config.GetResourceType(resource)
+		log.Debugf("Checking if provider, %s, supports %s and type, %s, pair.", p.Name(), resource.Provenance().Capability, resourceType)
+		for _, validType := range mapping {
+			if validType == resourceType {
+				resourceValid = true
```

Comment:
nit: you can just `return nil` at this point, and then you don't need `resourceValid` at all.